tiledb/sm/array_schema/dimension.h

```diff
@@ -113,6 +115,59 @@ class Dimension {
   /** Returns true if this is an anonymous (unlabled) dimension **/
   bool is_anonymous() const;
 
+  /**
+   * Computed the minimum bounding range of the values stored in
+   * `tile`.
+   */
+  void compute_mbr(const Tile& tile, Range* mbr) const;
+
+  /**
+   * Computed the minimum bounding range of the values stored in
+   * `tile`.
+   */
+  template <class T>
+  static void compute_mbr(const Dimension* dim, const Tile& tile, Range* mbr);
+
+  /**
+   * Crops the input 1D range such that it does not exceed the
+   * dimension domain.
+   */
+  void crop_range(Range* range) const;
+
+  /**
+   * Crops the input 1D range such that it does not exceed the
+   * input dimension domain.
+   */
+  template <class T>
+  static void crop_range(const Dimension* dim, Range* range);
+
+  /** Expand 1D range `r` using value `v`. */
+  void expand_range_v(const void* v, Range* r) const;
+
+  /** Expand 1D range `r` using value `v`. */
+  template <class T>
+  static void expand_range_v(const Dimension* dim, const void* v, Range* r);
+
+  /** Expand 1D range `r2` using 1D range `r1`. */
+  void expand_range(const Range& r1, Range* r2) const;
+
+  /** Expand 1D range `r2` using 1D range `r1`. */
+  template <class T>
+  static void expand_range(const Dimension* dim, const Range& r1, Range* r2);
+
+  /**
+   * Expands the input 1D range to coincide with the dimension tiles.
+   * It is a noop if the tile extents are null and for real domains.
```

Comment:
From `Dimension::set_tile_extent`:

```
Status Dimension::set_tile_extent(const void* tile_extent) {
  if (domain_ == nullptr)
    return Status::DimensionError(
        "Cannot set tile extent; Domain must be set first");

  // Note: this check was added in release 1.6.0. Older arrays may have been
  // serialized with a null extent, and so it is still supported internally.
  // But users can not construct dimension objects with null tile extents.
  if (tile_extent == nullptr)
    return LOG_STATUS(Status::DimensionError(
        "Cannot set tile extent; tile extent cannot be null"));
```

So it can be used internally, but only for old arrays for backwards compatibility.

Reply:
gotcha, thanks